tests/basic_fuzzer.cpp

```diff
@@ -18,62 +28,361 @@ namespace {
+        output.resize(expected_length);
+        size_t utf8_to_utf16 = implementation.convert_utf8_to_utf16le(
+            input.data(), input.size(),
+            reinterpret_cast<char16_t *>(output.data()));
+        ASSERT_TRUE(expected_length > 0 && expected_length == output.size() && expected_length == utf8_to_utf16);
+
+        expected_length = implementation.utf32_length_from_utf8(input.data(), input.size());
+        output.resize(expected_length);
+        size_t utf8_to_utf32 = implementation.convert_utf8_to_utf32(
+            input.data(), input.size(),
+            reinterpret_cast<char32_t *>(output.data()));
+        ASSERT_TRUE(expected_length > 0 && expected_length == output.size() && expected_length == utf8_to_utf32);
+      }
+      if (is_ok_utf16) {
+        size_t expected_length = implementation.utf8_length_from_utf16le(reinterpret_cast<char16_t *>(input.data()), input.size() / sizeof(char16_t));
+        output.resize(expected_length);
+        size_t utf16_to_utf8 = implementation.convert_utf16le_to_utf8(
+            reinterpret_cast<char16_t *>(input.data()),
+            input.size() / sizeof(char16_t), output.data());
+        ASSERT_TRUE(expected_length > 0 && expected_length == output.size() && expected_length == utf16_to_utf8);
+
+        expected_length = implementation.utf32_length_from_utf16le(reinterpret_cast<char16_t *>(input.data()), input.size() / sizeof(char16_t));
+        output.resize(expected_length);
+        size_t utf16_to_utf32 = implementation.convert_utf16le_to_utf32(
+            reinterpret_cast<char16_t *>(input.data()),
+            input.size() / sizeof(char16_t), reinterpret_cast<char32_t *>(output.data()));
+        ASSERT_TRUE(expected_length > 0 && expected_length == output.size() && expected_length == utf16_to_utf32);
+      }
+      if (is_ok_utf32) {
+        size_t expected_length = implementation.utf8_length_from_utf32(reinterpret_cast<char32_t *>(input.data()), input.size() / sizeof(char32_t));
+        output.resize(expected_length);
+        size_t utf32_to_utf8 = implementation.convert_utf32_to_utf8(
+            reinterpret_cast<char32_t *>(input.data()),
+            input.size() / sizeof(char32_t), output.data());
+        ASSERT_TRUE(expected_length > 0 && expected_length == output.size() && expected_length == utf32_to_utf8);
+
+        expected_length = implementation.utf16_length_from_utf32(reinterpret_cast<char32_t *>(input.data()), input.size() / sizeof(char32_t));
+        output.resize(expected_length);
+        size_t utf32_to_utf16 = implementation.convert_utf32_to_utf16le(
+            reinterpret_cast<char32_t *>(input.data()),
+            input.size() / sizeof(char32_t), reinterpret_cast<char16_t *>(output.data()));
+        ASSERT_TRUE(expected_length > 0 && expected_length == output.size() && expected_length == utf32_to_utf16);
+      }
+    }
+  }
+}
+
+
+int main(int argc, char *argv[]) { return simdutf::test::main(argc, argv); }
```

Comment:
We would like to be able to pass parameters so that you can effectively have this fuzzer run for a long time. One possibility would be to keep your program as-is but to pass a seed, so one could have this program with different seeds, we would be running different tests. So I could effectively have a script that runs this programs millions of times, with different seeds, and thus I get thorough testing (e.g., I could run this program for days or weeks).